### PR TITLE
Subscription, Queue and Topic creation no longer uses the container

### PR DESCRIPTION
**Pull request by danielmarbach**

### CODE REVIEW

**Comment by danielmarbach** [comment]
@Particular/azure-maintainers please review

File: src/Transport/Topology/Topologies/ForwardingTopology.cs [hunk 1]
```diff
@@ -17,8 +17,8 @@ class ForwardingTopologyInternal : ITopologyInternal
         ILog logger = LogManager.GetLogger("ForwardingTopology");
         ITopologySectionManagerInternal topologySectionManager;
         ITransportPartsContainerInternal container;
-        AzureServiceBusQueueCreator queueCreator;
-        AzureServiceBusTopicCreator topicCreator;
+        AzureServiceBusQueueCreator createQueues;
```
Comment (SeanFeldman) [inline src/Transport/Topology/Topologies/ForwardingTopology.cs]: revert name change

File: src/Transport/Topology/Topologies/ForwardingTopology.cs [hunk 2]
```diff
@@ -17,8 +17,8 @@ class ForwardingTopologyInternal : ITopologyInternal
...
+        AzureServiceBusTopicCreator createTopics;
```
Comment (SeanFeldman) [inline src/Transport/Topology/Topologies/ForwardingTopology.cs]: revert name change

File: src/Transport/Topology/Topologies/ForwardingTopology.cs [hunk 3]
```diff
@@ -43,10 +44,11 @@ internal ForwardingTopologyInternal(ITransportPartsContainerInternal container)
 
         public void Initialize(SettingsHolder settings)
         {
+            createQueues = new AzureServiceBusQueueCreator(Settings.QueueSettings, settings);
```
Comment (SeanFeldman) [inline src/Transport/Topology/Topologies/ForwardingTopology.cs]: Rather see these after `ApplyDefaults()` and `InitializeContainer()`.

File: src/Transport/Topology/TopologyCreator.cs [hunk 4]
```diff
@@ -5,12 +5,16 @@ namespace NServiceBus.Transport.AzureServiceBus
...
     class TopologyCreator : ICreateTopologyInternal
     {
-        ITransportPartsContainerInternal container;
         IManageNamespaceManagerLifeCycleInternal namespaces;
+        ICreateAzureServiceBusSubscriptionsInternal createSubscriptions;
+        AzureServiceBusQueueCreator createQueues;
+        AzureServiceBusTopicCreator createTopics;
 
-        public TopologyCreator(ITransportPartsContainerInternal container, IManageNamespaceManagerLifeCycleInternal namespaces)
+        public TopologyCreator(ICreateAzureServiceBusSubscriptionsInternal createSubscriptions, AzureServiceBusQueueCreator createQueues, AzureServiceBusTopicCreator createTopics, IManageNamespaceManagerLifeCycleInternal namespaces)
```
Comment (SeanFeldman) [inline src/Transport/Topology/TopologyCreator.cs]: `createQueues` -> `queuesCreator`
`createTopics` -> `topicsCreator`
`createSubscriptions` -> `subscriptionsCreator`

File: src/Transport/Topology/Topologies/EndpointOrientedTopology.cs [hunk 5]
```diff
@@ -39,10 +39,11 @@ internal EndpointOrientedTopologyInternal(ITransportPartsContainerInternal conta
 
         public void Initialize(SettingsHolder settings)
         {
+            createQueues = new AzureServiceBusQueueCreator(Settings.QueueSettings, settings);
```
Comment (danielmarbach) [inline src/Transport/Topology/Topologies/EndpointOrientedTopology.cs]: Actually it doesn't matter but happy to change it if you prefer it

Comment (danielmarbach) [inline src/Transport/Topology/Topologies/EndpointOrientedTopology.cs]: Yeah good point. Moved it too far

File: src/Transport/Topology/Topologies/ForwardingTopology.cs [hunk 6]
```diff
@@ -43,10 +44,11 @@ internal ForwardingTopologyInternal(ITransportPartsContainerInternal container)
 
         public void Initialize(SettingsHolder settings)
         {
+            queuesCreator = new AzureServiceBusQueueCreator(Settings.QueueSettings, settings);
```
Comment (SeanFeldman) [inline src/Transport/Topology/Topologies/ForwardingTopology.cs]: `ApplyDefaults(settings);` be first like in the `EndpointOrientedTopology`

**Comment by SeanFeldman** [comment]
👍 can be merged

**Comment by boblangley** [comment]
Merged #476

**Pull request merged**